src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const fromParts = fromFormat.slice(0, 3);
+  const toParts = toFormat.slice(0, 3);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromParts.length; i++) {
+    dateMap[fromParts[i]] = dateParts[i];
+  }
+
+  if (fromParts.includes('YYYY') && toParts.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (fromParts.includes('YY') && toParts.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
+  }
+
+  let newDate = '';
+
+  for (let i = 0; i < toParts.length; i++) {
+    if (i > 0) {
+      newDate += toSeparator;
+    }
+    newDate += dateMap[toParts[i]];
```

Comment:
The loop for constructing the `newDate` string is clear. However, you could use the array method `map` along with `join` to simplify this process. This would remove the need for the manual index checking and concatenation of the separator, making the code more declarative.